pkg/model/resources/nodeup.go

```diff
@@ -60,6 +60,22 @@ function ensure-install-dir() {
   cd ${INSTALL_DIR}
 }
 
+try-download-file() {
+  local -r url="$1"
+  local -r file="$2"
+  local -r auth="$3"
+  if [[ "$auth" == "" ]]; then
+		if curl -f --compressed -Lo "${file}" --connect-timeout 20 --retry 6 --retry-delay 10 "${url}"; then return 0; fi
+		if wget --compression=auto -O "${file}" --connect-timeout=20 --tries=6 --wait=10 "${url}"; then return 0; fi
+		if curl -f -Lo "${file}" --connect-timeout 20 --retry 6 --retry-delay 10 "${url}"; then return 0; fi
+		if wget -O "${file}" --connect-timeout=20 --tries=6 --wait=10 "${url}"; then return 0; fi
+  else
+		if curl -f --compressed -Lo "${file}" -v -H "Authorization: Bearer ${auth}" --connect-timeout 20 --retry 6 --retry-delay 10 "${url}"; then return 0; fi
```

Comment:
One of the things we have to be careful about here is logging the auth token in the output, in particularly for the e2e tests.  However the good news is that this script _doesn't_ have the `set -x` / `set -o xtrace` so we should be OK!